code/parachain/frame/cosmwasm/src/lib.rs

```diff
@@ -498,1084 +513,836 @@ pub mod pallet {
-					with_tag
-				})
-				.map_err(|_| ())
-		}
-
-		pub(crate) fn do_secp256k1_verify(
-			message_hash: &[u8],
-			signature: &[u8],
-			public_key: &[u8],
-		) -> bool {
-			let message_hash = match message_hash.try_into() {
-				Ok(message_hash) => message_hash,
-				Err(_) => return false,
-			};
-
-			// We are expecting 64 bytes long public keys but the substrate function use an
-			// additional byte for recovery id. So we insert a dummy byte.
-			let signature = {
-				let mut signature_inner = [0_u8; SUBSTRATE_ECDSA_SIGNATURE_LEN];
-				signature_inner[..SUBSTRATE_ECDSA_SIGNATURE_LEN - 1].copy_from_slice(signature);
-				ecdsa::Signature(signature_inner)
-			};
-
-			let public_key = match libsecp256k1::PublicKey::parse_slice(public_key, None) {
-				Ok(public_key) => ecdsa::Public::from_raw(public_key.serialize_compressed()),
-				Err(_) => return false,
-			};
-
-			sp_io::crypto::ecdsa_verify_prehashed(&signature, &message_hash, &public_key)
-		}
-
-		pub(crate) fn do_ed25519_batch_verify(
-			messages: &[&[u8]],
-			signatures: &[&[u8]],
-			public_keys: &[&[u8]],
-		) -> bool {
-			let mut messages = messages.to_vec();
-			let mut public_keys = public_keys.to_vec();
-
-			if messages.len() == signatures.len() && messages.len() == public_keys.len() {
-				// Nothing needs to be done
-			} else if messages.len() == 1 && signatures.len() == public_keys.len() {
-				// There can be a single message signed with different signature-public key pairs
-				messages = messages.repeat(signatures.len());
-			} else if public_keys.len() == 1 && messages.len() == signatures.len() {
-				// Single entity(with a public key) might wanna verify different messages
-				public_keys = public_keys.repeat(signatures.len());
-			} else {
-				// Any other case is wrong
-				return false
-			}
-
-			// Each batch verification process is started with `start_batch_verify` and ended with
-			// `finish_batch_verify`. When it is started, it needs to be properly finished. But this
-			// means `finish_batch_verify` will verify the previously pushed verification tasks. We
-			// converted all the public keys and signatures in-front not to unnecessarily verify
-			// previously pushed signatures. (Note that there is no function to ditch the batch
-			// verification early without doing any verification)
-			let mut verify_items = Vec::with_capacity(messages.len());
-			for ((message, signature), public_key) in
-				messages.iter().zip(signatures.iter()).zip(public_keys.iter())
-			{
-				match ((*signature).try_into(), (*public_key).try_into()) {
-					(Ok(signature), Ok(public_key)) =>
-						verify_items.push((signature, message, public_key)),
-					_ => return false,
-				}
-			}
+		// Else, the contract is not a pallet. We continue with the normal wasmi vm creation
+		// process.
+		let info = Self::contract_info(&contract)?;
+
+		// Reinstrument the code if needed
+		let code = Self::do_check_for_reinstrumentation(info.code_id)?.map_or(
```

Comment:
Why are we not doing all that in the cache miss branch just like the previous code? We now check for instrumentation and load from `InstrumentedCode` even if the code is already present in the cache?

Reply:
Ah yes you are correct. Forgot to fix the order. Will fix

Reply:
Fixed in d896f3567e1556f537608470dfa293682d0e71c0